### PR TITLE
use OS hostname in the Gateway clientID

### PR DESCRIPTION
**Pull request by stefanprodan**
Fix for #12

### CODE REVIEW

File: handler/handler.go [hunk 1]
```diff
@@ -22,7 +22,8 @@ func CreateNatsQueue(address string, port int) (*NatsQueue, error) {
 	natsURL := fmt.Sprintf("nats://%s:%d", address, port)
 	log.Printf("Opening connection to %s\n", natsURL)
 
-	clientID := "faas-publisher"
+	val, _ := os.Hostname()
+	clientID := "faas-publisher-" + val
```
Comment (alexellis) [inline handler/handler.go]: Should this be prefixed with faas-publisher? Have you tested os.Hostname() that it returns the container hostname and not localhost?

Comment (stefanprodan) [reply handler/handler.go]: `os.Hostname` inside a container will return the container id, it's the same as you did it here https://github.com/openfaas/nats-queue-worker/blob/master/queue-worker/main.go#L55

Comment (alexellis) [reply handler/handler.go]: Perfect, I thought that code looked familiar. 👍

**Comment by alexellis** [comment]
Thanks!

**Pull request merged**